packages/iov-keycontrol/src/keyring-entries/slip10.spec.ts

```diff
@@ -71,8 +87,77 @@ describe("Ed25519HdKeyringEntry", () => {
     }
   });
 
+  it("can create Secp256k1 identities", async () => {
+    const emptyEntries = [
+      // all possible ways to construct a Slip10KeyringEntry for Secp256k1
+      new Slip10KeyringEntry(emptySecp256k1Entry),
+      Slip10KeyringEntry.fromEntropyWithCurve(Slip10Curve.Secp256k1, Encoding.fromHex("51385c41df88cbe7c579e99de04259b1aa264d8e2416f1885228a4d069629fad")),
+      Slip10KeyringEntry.fromMnemonicWithCurve(Slip10Curve.Secp256k1, "execute wheel pupil bachelor crystal short domain faculty shrimp focus swap hazard"),
+    ];
+
+    for (const entry of emptyEntries) {
+      const newIdentity1 = await entry.createIdentityWithPath([Slip10RawIndex.hardened(0)]);
+      const newIdentity2 = await entry.createIdentityWithPath([Slip10RawIndex.hardened(1)]);
+      const newIdentity3 = await entry.createIdentityWithPath([Slip10RawIndex.hardened(1), Slip10RawIndex.hardened(0)]);
+
+      expect(newIdentity1.pubkey.data).not.toEqual(newIdentity2.pubkey.data);
+      expect(newIdentity2.pubkey.data).not.toEqual(newIdentity3.pubkey.data);
+      expect(newIdentity3.pubkey.data).not.toEqual(newIdentity1.pubkey.data);
+
+      const identities = entry.getIdentities();
+      expect(identities.length).toEqual(3);
+      expect(identities[0].pubkey.algo).toEqual(Algorithm.SECP256K1);
+      expect(identities[0].pubkey.data).toEqual(newIdentity1.pubkey.data);
+      expect(identities[1].pubkey.algo).toEqual(Algorithm.SECP256K1);
+      expect(identities[1].pubkey.data).toEqual(newIdentity2.pubkey.data);
+      expect(identities[2].pubkey.algo).toEqual(Algorithm.SECP256K1);
+      expect(identities[2].pubkey.data).toEqual(newIdentity3.pubkey.data);
+    }
+  });
+
+  it("generates secp256k1 keys compatible to other tools", async () => {
+    // Test data generated by the BIP44 tool of
+    // https://iancoleman.io/bip39/#english
+
+    const entry = Slip10KeyringEntry.fromMnemonicWithCurve(Slip10Curve.Secp256k1, "mushroom faint dumb venture million true skull grab pitch mesh share tortoise");
```

Comment:
Okay, so the same class can be used, just exposes one curve/mneumonic pair.
If I wanted the same mneumonic with two different curves, I would create two instances of this class, correct?

This is good. All the hard coding is done once, we can just wrap it up different ways.

Reply:
Correct, every Slip10KeyringEntry has exactly one fixed mnemonic and exactly one fixed curve type